v2/tools/generator/internal/codegen/storage/group_conversion_graph_builder.go

```diff
@@ -6,146 +6,79 @@
+	for name := range names {
+		b.Add(name)
 	}
-
-	result := &GroupConversionGraph{
-		group: b.group,
-		links: b.links,
-	}
-
-	return result, nil
 }
 
-// compatibilityReferencesConvertForward links any compatibility references forward to the following version
-func (b *GroupConversionGraphBuilder) compatibilityReferencesConvertForward(refs []astmodel.PackageReference) {
-	for i, ref := range refs {
-		if !b.isCompatibilityPackage(ref) {
-			continue
+// Build connects all the provided API definitions together into a single conversion graph
+func (b *GroupConversionGraphBuilder) Build() (*GroupConversionGraph, error) {
+	subgraphs := make(map[string]*ResourceConversionGraph, len(b.subBuilders))
+	for group, builder := range b.subBuilders {
+		subgraph, err := builder.Build()
+		if err != nil {
+			return nil, errors.Wrapf(err, "building subgraph for group %s", group)
 		}
 
-		// Safe to use i+1 because compatibility references will always be preceded by a reference to the original
-		// storage package reference
-		b.links[ref] = refs[i+1]
-	}
-}
-
-// apiReferencesConvertToStorage links each API type to the associated storage package
-func (b *GroupConversionGraphBuilder) apiReferencesConvertToStorage(refs []astmodel.PackageReference) {
-	for _, ref := range refs {
-		if s, ok := ref.(astmodel.StoragePackageReference); ok {
-			b.links[s.Local()] = s
-		}
+		subgraphs[group] = subgraph
 	}
-}
 
-// previewReferencesConvertBackward links each preview version to the immediately prior version, no matter whether it's
-// preview or GA.
-func (b *GroupConversionGraphBuilder) previewReferencesConvertBackward(refs []astmodel.PackageReference) {
-	for i, ref := range refs {
-		if i == 0 || !ref.IsPreview() {
-			continue
-		}
-
-		b.links[ref] = refs[i-1]
+	result := &GroupConversionGraph{
+		group:     b.group,
+		subGraphs: subgraphs,
 	}
-}
 
-// nonPreviewReferencesConvertForward links each version with the immediately following version.
-// By the time we run this stage, we should only have non-preview (aka GA) releases left
-func (b *GroupConversionGraphBuilder) nonPreviewReferencesConvertForward(refs []astmodel.PackageReference) {
-	for i, ref := range refs {
-		// Links are created from the current index to the next;
-		// if we're at the end of the sequence, there's nothing to do.
-		if i+1 >= len(refs) {
-			break
-		}
-
-		next := refs[i+1]
-		b.links[ref] = next
-	}
+	return result, nil
 }
 
-// withoutLinkedReferences returns a new slice of references, omitting any that are already linked into our graph
-func (b *GroupConversionGraphBuilder) withoutLinkedReferences(
-	refs []astmodel.PackageReference) []astmodel.PackageReference {
-	var result []astmodel.PackageReference
-	for _, ref := range refs {
-		if _, ok := b.links[ref]; !ok {
-			result = append(result, ref)
-		}
+// getSubBuilder finds the relevant builder for the group of the provided reference, creating one if necessary
```

Comment:
```suggestion
// getSubBuilder finds the relevant builder for the resource of the provided reference, creating one if necessary
```

?? Or am I misunderstanding what this is doing?